### DISCUSSION TITLE
about the contact algorithm

### DISCUSSION THREAD

**Discussion by hekungeo** [Anura3D/Anura3D_OpenSource#8] [Q&A] [answered]
Hi, 

I am wondering if the contact properties setting in GID can perform the multi-sliding surface moving in a landslide, something like retrogressive moving. Here is the model.
![image](https://user-images.githubusercontent.com/105496180/170969303-5b66d311-b495-4b3a-b197-c7e5b1bbc53c.png)

1. I have tried to define the surface as follows:
![image](https://user-images.githubusercontent.com/105496180/170969913-e8e89e28-e130-48d7-8efe-17ed0afe95fb.png)
 
However, errors occurred when I proceeded>
![image](https://user-images.githubusercontent.com/105496180/170970152-456f340b-eb1b-45ca-a28d-4249633d4d21.png)

2. Also, I have tried to define the contact surface according to the sliding blocks tutorial manual, as follows:
![image](https://user-images.githubusercontent.com/105496180/170972392-66aa2ad3-b08d-4765-9203-dae18d8e86dc.png)

but just Pause when I proceeded:

![image](https://user-images.githubusercontent.com/105496180/170972625-2bd86661-bccd-4b79-af75-606266fef258.png)

3. Additional, can the contact algorithm simulate the joint sets in rockslide simulation? 

Thanks in advance.

Best,
Kun

**Comment by ayerrocolom** [comment, marked as answer]
With the contact algorithm implemented in Anura3D, you can simulate the sliding of several deformable materials on top of a rigid material, in a way that the contact surface is always defined by the same nodes. So, in your problem, the contact algorithm can be defined between the different sliding wedges and the bottom material (in yellow in your first figure); similar to the sliding blocks problem in the tutorial manual. You cannot use the contact algorithm between different sliding wedges or different joint sets. I hope this makes sense. 

From your second error, it seems that there is something wrong with the contact information in the GOM file. I would recommend you to debug the code. I also see that you are using unsaturated materials. I would try dry or saturated materials first to check that the contact algorithm is properly applied. The combination of 2phase+suction formulation with contact algorithm should work but just to double-check to make sure there are no bugs.

**Reply by hekungeo** [reply]
Thank you so much. I will have a try.

**Reply by zhangxiaopengY** [reply]
Hi，Kun，have you solve the problem, I have the similar problem when modelling saturated material, can you share your experience, thanks.

**Reply by zhangxiaopengY** [reply]
I  also had problems with floating invalid.
![Snipaste_2022-06-09_16-42-15](https://user-images.githubusercontent.com/63776996/172805306-2bed021d-bb43-4e07-b9e4-e67388ffa38a.jpg)

**Reply by hekungeo** [reply]
> I also had problems with floating invalid. ![Snipaste_2022-06-09_16-42-15](https://user-images.githubusercontent.com/63776996/172805306-2bed021d-bb43-4e07-b9e4-e67388ffa38a.jpg)

Hi xiaopeng, 
Unfortunately, not yet now. I have tried the sliding blocks in tutorial manual, it works well. But in my case, the contact didn't work. Maybe we should do some tests from simple to complex models to debug it.

**Reply by zhangxiaopengY** [reply]
OK，I will try, thank U.

**Reply by ZeidBit** [reply]
I am also getting the same as your second error. I just have 4 materials with defined contact properties sliding over a rigid homogenous material. 
![image](https://user-images.githubusercontent.com/103214089/179624311-5fcb37ca-cea8-408c-9b9c-abc9915f807e.png)

Any idea what is going on? Seems to stop from reading the contact data from the GOM file..
